pkg/agent/agent_test.go

```diff
@@ -708,3 +712,263 @@ func TestRestorePortConfigs(t *testing.T) {
+			mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+			initializer := newAgentInitializer(mockOVSBridgeClient, nil)
+			tt.expectedCalls(mockOVSBridgeClient)
+
+			err := initializer.setOVSDatapath()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func mockIPsecPSKEnv(name string) func() {
+	_ = os.Setenv(ipsecPSKEnvKey, name)
+	return func() { os.Unsetenv(ipsecPSKEnvKey) }
+}
+
+func TestReadIPSecPSK(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec PSK env variable not set",
+			expectedErr: "IPsec PSK environment variable 'ANTREA_IPSEC_PSK' is not set or is empty",
+		},
+		{
+			name: "IPsec PSK env variable set",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{
+				networkConfig: &config.NetworkConfig{
+					IPsecConfig: config.IPsecConfig{},
+				},
+			}
+			if tt.expectedErr == "" {
+				defer mockIPsecPSKEnv("key")()
+			}
+
+			err := initializer.readIPSecPSK()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetNodeConfig(t *testing.T) {
+	nodeConfig := &config.NodeConfig{}
+	initializer := &Initializer{
+		nodeConfig: nodeConfig,
+	}
+	assert.Equal(t, nodeConfig, initializer.GetNodeConfig())
+}
+
+func TestWaitForIPSecMonitorDaemon(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec monitor is not running",
+			expectedErr: "IPsec was requested, but the OVS IPsec monitor does not seem to be running",
+		},
+		{
+			name: "IPsec monitor running",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{}
+			if tt.expectedErr == "" {
+				appFS := afero.NewMemMapFs()
+				err := appFS.MkdirAll("/var/run/openvswitch", 0777)
+				require.NoError(t, err)
+				_, err = appFS.Create("/var/run/openvswitch/ovs-monitor-ipsec.pid")
+				require.NoError(t, err)
+				appFs = appFS
+				defer func() {
+					appFs = afero.NewOsFs()
+				}()
+			}
+
+			err := initializer.waitForIPsecMonitorDaemon()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, initializer.waitForIPsecMonitorDaemon(), tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestInitVMLocalConfig(t *testing.T) {
+	ipDevice := &net.Interface{
+		Name: "fakeUplinkInterface",
+	}
+	testNode := &crdv1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "testNode", Namespace: "external"},
+		Spec: crdv1alpha1.ExternalNodeSpec{
+			Interfaces: []crdv1alpha1.NetworkInterface{
+				{
+					IPs: []string{"192.168.1.2"},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name        string
+		nodeName    string
+		client      *fake.Clientset
```

Comment:
I didn't see this is defined in any test cases.

Reply:
removed, thanks